### PR TITLE
Script updated to validate the result of pipeline run

### PR DESCRIPTION
**Pull request by ppitonak**
The script now no only deploys and runs the pipeline but also checks its results.

Outside of scope of this PR: #60

### CODE REVIEW

File: demo.sh [hunk 1]
```diff
@@ -47,19 +46,29 @@ TKN() {
...
   tkn version >/dev/null 2>&1 || err 1 "no tkn binary found"
-  oc version >/dev/null 2>&1 || err 1 "no oc binary found"
+  oc version --client >/dev/null 2>&1 || err 1 "no oc binary found"
   return 0
 }
 
-declare -r TASKS_DIR=tmp/tasks
+demo.validate_pipelinerun() {
+  local failed=0
+  results=$(oc get pipelinerun.tekton.dev -n ${NAMESPACE} --output=jsonpath='{range .items[*]}{.metadata.name}={.status.conditions[*].type}{.status.conditions[*].status}{" "}{end}')
```
Comment (sthaha) [inline demo.sh]: local results

File: demo.sh [hunk 2]
```diff
@@ -47,19 +46,29 @@ TKN() {
...
   tkn version >/dev/null 2>&1 || err 1 "no tkn binary found"
-  oc version >/dev/null 2>&1 || err 1 "no oc binary found"
+  oc version --client >/dev/null 2>&1 || err 1 "no oc binary found"
   return 0
 }
 
-declare -r TASKS_DIR=tmp/tasks
+demo.validate_pipelinerun() {
+  local failed=0
+  local results=$(oc get pipelinerun.tekton.dev -n ${NAMESPACE} --output=jsonpath='{range .items[*]}{.metadata.name}={.status.conditions[*].type}{.status.conditions[*].status}{" "}{end}')
```
Comment (sthaha) [inline demo.sh]: please always quote "" variables e.g. `-n "$NAMESPACE" `

Comment (sthaha) [reply demo.sh]: you can make use of arrays to iterate ...  might want to consider something like this 

```
loca results=( $(oc get pipelineruns --template='
{{range .items -}}
  {{ $pr := .metadata.name -}}
  {{ $c := index .status.conditions 0 -}}
  {{ $pr }}={{ $c.type }}{{ $c.status }}
{{ end }}
') )

for pr in ${result[@]}; do echo $pr ... ; done  

```

Comment (ppitonak) [reply demo.sh]: Alternatively, I can just extract the jsonpath into separate variable, which one do you prefer?

**Review by sthaha — APPROVED** [review]
/lgtm

**Pull request merged**